### PR TITLE
54 port arrival time minus

### PR DESCRIPTION
**Pull request by ckulfan**
Added processing for the arrival time ETA, which is calculated as the number of seconds remaining until the boat is predicted to get to dock. WSDOT sometime returns ETAs in the past, when the boat is running late because the ETA value is not recalculated during the voyage. We just set this to zero to avoid thrashing the results as it gets to dock. The ETA can also be slow to getting set when a boat departs, in this case the ETA from WSDOT is null which results in us setting the ETA to zero until we get a result. 

I also missed a logging entry in FerryTempo.js, which is fixed here in this pull request.

Addresses issue number 54.

### CODE REVIEW

File: src/FerryTempo.js [hunk 1]
```diff
@@ -90,8 +94,27 @@ export default {
           boatDelay = epochTimeStamp - epochScheduledDeparture;
         }
 
+        // Calculate the BoatETA as a function of the progress we have made on the journey. We use this value unless we do not get
```
Comment (alexmarc-us) [inline src/FerryTempo.js:97]: Nit: format multi-line comments with `/**`
```suggestion
        /** Calculate the BoatETA as a function of the progress we have made on the journey. We use this value unless we do not get
        *   a departureTime value from WSDOT for the boat, in wich case we will use the arrivalTimeEta computed from the Eta field and now().
        **/
```

File: src/FerryTempo.js [hunk 2]
```diff
@@ -90,8 +94,27 @@ export default {
...
+        let arrivalTimeEta = 0;
+        if (InService && !AtDock && epochEta != 0 ) {
+          let departureTime = getEpochSecondsFromWSDOT(LeftDock);
+          if (departureTime != 0) {
+            arrivalTimeEta = Math.trunc((epochEta - departureTime) * (1.0 - getProgress(routeData, currentLocation)));
+          } else {
+            if (epochEta > getCurrentEpochSeconds()) {
+              arrivalTimeEta = epochEta - getCurrentEpochSeconds();
+            } else if (epochEta != 0) {
+              // ETA is negative typically when the boat is late arriving since the ETA doesn't get recalculated by WSDOT.
```
Comment (alexmarc-us) [inline src/FerryTempo.js:108]: Same multi-line nit here.

File: src/Utils.js [hunk 3]
```diff
@@ -50,6 +50,24 @@ export function getCurrentEpochSeconds() {
   return Math.floor(Date.now() / 1000);
 }
 
+/** 
+ * Return the provided epoch time value into a string for use in logging and debugging
+ * @return {string} The input epoch time converted to human readable format
+ */
+export function getHumanDateFromEpochSeconds(epochSec) {
+  function pad(n) {return n<10 ? '0'+n : n}
```
Comment (alexmarc-us) [inline src/Utils.js:58]: It's not clear to me what `pad` does. Could you add a comment with an example?

File: test/Utils.test.js [hunk 4]
```diff
@@ -66,3 +67,12 @@ describe('calculateDistance function', () => {
     expect(calculateDistance(coord1, coord2)).toBeCloseTo(559.1, 1);
   });
 });
+
+describe('getHumanDateFromEpochSeconds function', () => {
```
Comment (alexmarc-us) [inline test/Utils.test.js:71]: Do we need a test for the `pad` case?

**Review by alexmarc-us — CHANGES_REQUESTED** [review]

File: src/FerryTempo.js (hunk 2 above)
Comment (ckulfan) [reply src/FerryTempo.js:110]: should we create a lint rule for this?

File: test/Utils.test.js (hunk 4 above)
Comment (ckulfan) [reply test/Utils.test.js:71]: It is tested in the second result. 5->05; 7->07 while 10 and 51 are unaffected

**Review by alexmarc-us — APPROVED** [review]

**Pull request merged**